File: Apps/iOS7/Segue/Segue/SegueViewController.m
//
//  SegueViewController.m
//  Segue
//
//  Created by Clemens Wagner on 24.07.13.
//  Copyright (c) 2013 Cocoaneheads. All rights reserved.
//

#import "SegueViewController.h"
#import "ResultViewController.h"
#import "Cube.h"

@interface SegueViewController ()

@property (weak, nonatomic) IBOutlet UIDatePicker *datePicker;
@property (strong, nonatomic) IBOutlet Cube *cube;

@end

@implementation SegueViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    NSLog(@"Cube: length = %.f, color = %@", self.cube.length, self.cube.color);
}

- (void)prepareForSegue:(UIStoryboardSegue *)inSegue sender:(id)inSender {
    if([inSegue.identifier isEqualToString:@"dialog"]) {
        ResultViewController *theController = inSegue.destinationViewController;

        theController.birthDate = self.datePicker.date;
    }
}

@end

File: Apps/iOS8/Swift/Segue/Segue/SegueViewController.swift
//
//  SegueViewController.swift
//  Segue
//
//  Created by Clemens Wagner on 23.06.14.
//  Copyright (c) 2014 Cocoaneheads. All rights reserved.
//

import UIKit

class SegueViewController: UIViewController {
    @IBOutlet var datePicker : UIDatePicker
    @IBOutlet var cube : Cube

    override func viewDidLoad() {
        super.viewDidLoad()
        NSLog("Cube: length = %.f, color = %@", self.cube.length, self.cube.color!)
    }
    
    override func prepareForSegue(inSegue: UIStoryboardSegue!, sender inSender: AnyObject!) {
        if(inSegue.identifier == "dialog") {
            let theController = inSegue.destinationViewController as ResultViewController
            
            theController.birthDate = self.datePicker.date
        }
    }
}
